src/tools/manifest2proto.ts

```diff
@@ -7,30 +7,200 @@
  * subject to an additional IP rights grant found at
  * http://polymer.github.io/PATENTS.txt
  */
-import {Runtime} from '../runtime/runtime.js';
 import protobuf from 'protobufjs';
+import {Runtime} from '../runtime/runtime.js';
+import {Recipe} from '../runtime/recipe/recipe.js';
+import {Handle} from '../runtime/recipe/handle.js';
+import {Particle} from '../runtime/recipe/particle.js';
+import {Type, CollectionType, ReferenceType} from '../runtime/type.js';
+import {Schema} from '../runtime/schema.js';
+import {ParticleSpec} from '../runtime/particle-spec.js';
+import {assert} from '../platform/assert-web.js';
+import {findLongRunningArcId} from './storage-key-recipe-resolver.js';
+import {Manifest} from '../runtime/manifest.js';
+import {Capabilities} from '../runtime/capabilities.js';
 
+// Import Proto declaration.
 const rootNamespace = protobuf.loadSync('./java/arcs/core/data/proto/recipe.proto');
-const envelope = rootNamespace.lookupType('arcs.RecipeEnvelopeProto');
+const manifestProto = rootNamespace.lookupType('arcs.ManifestProto');
+const FATE_ENUM = rootNamespace.lookupEnum('arcs.Fate');
+const CAPABILITY_ENUM = rootNamespace.lookupEnum('arcs.Capability');
+const DIRECTION_ENUM = rootNamespace.lookupEnum('arcs.Direction');
+const PRIMITIVE_TYPE_ENUM = rootNamespace.lookupEnum('arcs.PrimitiveTypeProto');
 
 export async function serialize2proto(path: string): Promise<Uint8Array> {
   const manifest = await Runtime.parseFile(path);
 
-  if (manifest.recipes.length !== 1) throw Error('Manifest should have exactly one recipe');
+  if (manifest.imports.length) {
+    throw Error('Only single-file manifests are currently supported');
+  }
+
+  const json = await manifestToJSON(manifest);
+
+  const error = manifestProto.verify(json);
+  if (error) throw Error(error);
 
-  // This is a super early sketch of a plan serialization, just for demo purposes.
-  const payload = {
-    recipe: manifest.recipes.map(r => ({
-      name: r.name,
-      particles: r.particles.map(p => ({specName: p.name})),
-      handles: r.handles.map(h => ({name: h.localName})),
-    }))[0],
+  return manifestProto.encode(manifestProto.create(json)).finish();
+}
+
+export async function manifestToJSON(manifest: Manifest) {
```

Comment:
This isn't quite JSON, it's more like an object... But I don't think manifestToObject is a great name, probably manifestToJSON is still better.. Might be worth mentioning something about this in a comment. Probably just once, you don't need to do it for every function.

Reply:
Change to protoPayload, which is how the protobufjs call is.